airflow/docker_image.go

```diff
@@ -150,6 +168,109 @@ func (d *DockerImage) Pytest(pytestFile, airflowHome, envFile string, pytestArgs
 	return outb.String(), docErr
 }
 
+func (d *DockerImage) ConflictTest(workingDirectory, testHomeDirectory string, buildConfig airflowTypes.ImageBuildConfig) (string, error) {
+	dockerCommand := config.CFG.DockerCommand.GetString()
+	// delete container
+	err := cmdExec(dockerCommand, nil, nil, "rm", "astro-temp-container")
+	if err != nil {
+		log.Debug(err)
+	}
+	// Change to location of Dockerfile
+	err = os.Chdir(buildConfig.Path)
+	if err != nil {
+		return "", err
+	}
+	args := []string{
+		"build",
+		"-t",
+		"conflict-check:latest",
+		"-f",
+		"conflict-check.Dockerfile",
+		".",
+	}
+
+	// // Run the Docker build command
+	// cmd := exec.Command(dockerCommand, args...)
+
+	// Create a buffer to capture the command output
+	var stdout, stderr bytes.Buffer
+	multiStdout := io.MultiWriter(&stdout, os.Stdout)
+	multiStderr := io.MultiWriter(&stderr, os.Stdout)
+
+	// Start the command execution
+	err = cmdExec(dockerCommand, multiStdout, multiStderr, args...)
+	if err != nil {
+		return "", err
+	}
+	// Wait for the command to complete
+	// err = cmd.Wait()
+	// Get the exit code
+	exitCode := ""
+	if _, ok := err.(*exec.ExitError); ok {
+		// The command exited with a non-zero status
+		exitCode = parseExitCode(stderr.String())
+	} else if err != nil {
+		// An error occurred while running the command
+		return "", err
+	}
+	// Run a temporary container to copy the file from the image
+	err = cmdExec(dockerCommand, nil, nil, "create", "--name", "astro-temp-container", "conflict-check:latest")
+	if err != nil {
+		return exitCode, err
+	}
+	// Copy the result.txt file from the container to the destination folder
+	err1 := cmdExec(dockerCommand, nil, nil, "cp", "astro-temp-container:/usr/local/airflow/conflict-test-results.txt", "./"+testHomeDirectory)
+	if err1 != nil {
+		// Remove the temporary container
+		err = cmdExec(dockerCommand, nil, nil, "rm", "astro-temp-container")
```

Comment:
nit: similar thing, this could be moved to defer statement, right after create call